llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -6251,12 +6251,13 @@ static OverflowResult mapOverflowResult(ConstantRange::OverflowResult OR) {
 
 /// Combine constant ranges from computeConstantRange() and computeKnownBits().
 static ConstantRange computeConstantRangeIncludingKnownBits(
-    const Value *V, bool ForSigned, const DataLayout &DL, AssumptionCache *AC,
-    const Instruction *CxtI, const DominatorTree *DT,
+    const CachedBitsConstValue &V, bool ForSigned, const DataLayout &DL,
+    AssumptionCache *AC, const Instruction *CxtI, const DominatorTree *DT,
     bool UseInstrInfo = true) {
-  KnownBits Known =
-      computeKnownBits(V, DL, /*Depth=*/0, AC, CxtI, DT, UseInstrInfo);
-  ConstantRange CR1 = ConstantRange::fromKnownBits(Known, ForSigned);
+  ConstantRange CR1 = ConstantRange::fromKnownBits(
+      V.getKnownBits(SimplifyQuery(DL, /*TLI*/ nullptr, DT, AC,
+                                   safeCxtI(V, CxtI), UseInstrInfo)),
+      ForSigned);
```

Comment:
You might just add an API for `getCR()` in the same vein is `getKnownBits` For now it can just do the above logic, but we might want to cache that someday as well. Also seems to fit.